### PR TITLE
Unapologetic Skrell Buff

### PR DESCRIPTION
**Pull request by ParadoxSpace**
Thanks a lot to MoondancerPony.
Adds a psionic ping ability to Skrell and Armalis, which allows them to detect nearby lifeforms (except machines, vaurca, and diona) in a non-specific fashion.

image: https://i.imgur.com/0hpaskI.png

### CODE REVIEW

**Comment by ParadoxSpace** [comment]
!wip

**Comment by Arrow768** [comment]
Screenshots please

**Comment by ParadoxSpace** [comment]
https://i.imgur.com/F6xrz6A.png

**Comment by ParadoxSpace** [comment]
comma is gone

File: code/modules/mob/living/carbon/human/human_powers.dm [hunk 1]
```diff
@@ -1019,3 +1022,64 @@
 
 		to_chat(src, output)
 
+/mob/living/carbon/human
+	var/next_sonar_ping = 0
```
Comment (skull132) [inline code/modules/mob/living/carbon/human/human_powers.dm]: Use the `last_special` variable.

File: code/modules/mob/living/carbon/human/human_powers.dm [hunk 2]
```diff
@@ -1019,3 +1022,64 @@
...
+		return
+	next_sonar_ping += 10 SECONDS
+	var/heard_something = FALSE
+	to_chat(src, "<span class='notice'>You take a moment to tune into the local Nlom...</span>")
+	var/dirs = list()
+	for(var/mob/living/L in range(20))
+		var/turf/T = get_turf(L)
+		if(!T || L == src || L.stat == DEAD || L.isSynthetic() || L.is_diona() || isvaurca(L) || L.invisibility == INVISIBILITY_LEVEL_TWO)
+			continue
+		heard_something = TRUE
```
Comment (skull132) [inline code/modules/mob/living/carbon/human/human_powers.dm]: `heard_something` is the same as `dirs.len`. Don't need it.

File: code/modules/mob/living/carbon/human/human_powers.dm [hunk 3]
```diff
@@ -1019,3 +1022,64 @@
...
+					dist = "farther away"
+				else
+					dist = "far away"
+		else
+			dist = "on top of you"
+		LAZYINITLIST(dirs[direction])
+		dirs[direction][dist] += 1
+	for(var/d in dirs)
+		var/feedback = list()
+		if(isnull(d)) // can be 0
```
Comment (skull132) [inline code/modules/mob/living/carbon/human/human_powers.dm]: Is this check even necessary? A d doesn't get added if there's nothing to add to it.

File: code/modules/mob/living/carbon/human/human_powers.dm [hunk 4]
```diff
@@ -1019,3 +1022,64 @@
...
+			continue
+		for(var/dst in dirs[d])
+			if(isnull(dst))
```
Comment (skull132) [inline code/modules/mob/living/carbon/human/human_powers.dm]: again is this even necessary?

File: code/modules/mob/living/carbon/human/human_powers.dm [hunk 5]
```diff
@@ -1019,3 +1022,64 @@
...
+				continue // don't need to print if there's nothing there
+			feedback += "[dirs[d][dst]] psionic signature\s [dst],"
+		if(LAZYLEN(feedback) > 1)
```
Comment (skull132) [inline code/modules/mob/living/carbon/human/human_powers.dm]: Feedback is not a lazy list. Use `feedback.len`

File: code/modules/mob/living/carbon/human/human_powers.dm [hunk 6]
```diff
@@ -1019,3 +1022,64 @@
...
+			feedback[LAZYLEN(feedback)-1] += " and"
```
Comment (skull132) [inline code/modules/mob/living/carbon/human/human_powers.dm]:
```suggestion
			feedback[feedback.len - 1] += " and"
```

File: code/modules/mob/living/carbon/human/human_powers.dm (hunk 3 above)
Comment (MarinaGryphon) [reply code/modules/mob/living/carbon/human/human_powers.dm]: It's associative, and it's to differentiate it from 0, iirc, but maybe you're right.

File: code/modules/mob/living/carbon/human/human_powers.dm (hunk 5 above)
Comment (MarinaGryphon) [reply code/modules/mob/living/carbon/human/human_powers.dm]:
```suggestion
		if(feedback.len > 1)
```

File: code/modules/mob/living/carbon/human/human_powers.dm [hunk 7]
```diff
@@ -1019,3 +1022,64 @@
...
+		to_chat(src, span("notice", "You sense " + jointext(feedback, " ") + " towards the [dir2text(text2num(d))]."))
+	if(!heard_something)
```
Comment (MarinaGryphon) [inline code/modules/mob/living/carbon/human/human_powers.dm]:
```suggestion
	if(!dirs.len)
```

File: code/modules/mob/living/carbon/human/human_powers.dm (hunk 4 above)
Comment (MarinaGryphon) [reply code/modules/mob/living/carbon/human/human_powers.dm]: It's also here to differentiate between null and zero, but I'm not sure a null can actually be introduced. Probably just paranoia on my part.

**Comment by ParadoxSpace** [comment]
have mercy

**Comment by ParadoxSpace** [comment]
!review

**Pull request merged**